src/bacnet/basic/object/netport.c

```diff
@@ -720,12 +720,21 @@ bool Network_Port_MAC_Address(
     uint32_t object_instance, BACNET_OCTET_STRING *mac_address)
```

Comment:
Or maybe this:  bool Network_Port_MAC_Address(
      uint32_t object_instance, BACNET_OCTET_STRING *mac_address)
  {
     bool status = false;
  
      if (mac_address) {
         uint8_t mac_tmp[4 + 2] = { 0 }; 
         uint8_t mac_len = Network_Port_MAC_Address_Value(object_instance, mac_tmp, sizeof(mac_tmp));
 
         status = octetstring_init(mac_address, mac_tmp, mac_len);
      }  
  
     return status;
  }